actionview/test/template/tag_helper_test.rb

```diff
@@ -122,20 +122,15 @@ def test_tag_builder_do_not_modify_html_safe_options
   end
 
   def test_tag_with_dangerous_name
-    assert_equal "<#{"_" * COMMON_DANGEROUS_CHARS.size} />",
-                 tag(COMMON_DANGEROUS_CHARS)
-
-    assert_equal "<#{COMMON_DANGEROUS_CHARS} />",
-                 tag(COMMON_DANGEROUS_CHARS, nil, false, false)
+    assert_raise(ArgumentError) do
+      tag(COMMON_DANGEROUS_CHARS)
+    end
```

Comment:
I don't think this test is sufficient to exercise the regular expression, and I also think the COMMON_DANGEROUS_CHARS is the right set of characters to be testing.

I think this test should be something like this instead:

```suggestion
    INVALID_TAG_CHARS.each_char do |char|
      tag_name = "asdf-#{char}"
      assert_raise(ArgumentError, "expected #{tag_name.inspect} to be invalid") do
        tag(tag_name)
      end
    end
```

where

```ruby
   # these are the only chars in COMMON_DANGEROUS_CHARS that are invalid in a tag
   INVALID_TAG_CHARS = "> /"
```